File: BooksApp/src/app/_models/book.ts
import { Book_Author } from "./author.model";
import { Image } from "./image";

export class BookClass {
  bookId: number = 0;
  bookTitle: string = '';
  bookYear: number = 0;
  authorName: string = '';
  ownerId: number = 0;
   book_Authors: Book_Author[] = [];
  images: Image;
}

export class RentBookClass {
  bookId: number = 0;
  renterId: number = 0;
  ownerId: number = 0;
}
export class UnRentBookClass {
  bookId: number = 0;
  renterId: number = 0;
  ownerId: number = 0;
}

export interface Book {
  bookId: number;
  bookTitle: string;
  bookYear: number;
  authorName: string;
  ownerId: number;
  images: Image;
  book_Authors: Book_Author[];

}
export class GetOwnedBooks {
    bookId: number = 0;
    ownerId: number = 0;
    renterId: number = 0;
    bookTitle: string ="";
    bookYear: number = 0;
    book_Authors: Book_Author[] = [];
    authorName: string = "";
    images: Image;

}
export class BookUpdateClass{
  bookId: number = 0;
  bookTitle: string = "";
  bookYear: number = 0;

}

File: Books/Models/Book.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Books.Models
{
    public class Book
    {
        [Key]
        public int BookId { get; set; }

        [Column(TypeName = "nvarchar(100)")]
        public string BookTitle { get; set; }

        [Column(TypeName = "nvarchar(4)")]
        public int BookYear { get; set; }
        public AppUser Renter { get; set; }
        public int? RenterId { get; set; }
        public int OwnerId { get; set; }
        public AppUser Owner { get; set; }
        public List<Book_Author> Book_Authors { get; set; }
        public Image Images { get; set; }

    }
}
